charts/opentelemetry-kube-stack/templates/collector.yaml

```diff
@@ -0,0 +1,241 @@
+  {{- end }}
+  {{- if $collector.hostNetwork }}
+  hostNetwork: {{ $collector.hostNetwork }}
+  {{- end }}
+  {{- if $collector.shareProcessNamespace }}
+  shareProcessNamespace: {{ $collector.shareProcessNamespace }}
+  {{- end }}
+  {{- if $collector.priorityClassName }}
+  priorityClassName: {{ $collector.priorityClassName }}
+  {{- end }}
+  {{- if $collector.terminationGracePeriodSeconds }}
+  terminationGracePeriodSeconds: {{ $collector.terminationGracePeriodSeconds }}
+  {{- end }}
+  {{- with $collector.resources }}
+  resources:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.nodeSelector }}
+  nodeSelector:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.args }}
+  args:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.autoscaler }}
+  autoscaler:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.podDisruptionBudget }}
+  podDisruptionBudget:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.securityContext }}
+  securityContext:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.podSecurityContext }}
+  podSecurityContext:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.podAnnotations }}
+  podAnnotations:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.targetAllocator }}
+  targetAllocator:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.ingress }}
+  ingress:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.affinity }}
+  affinity:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.lifecycle }}
+  lifecycle:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.livenessProbe }}
+  livenessProbe:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.observability }}
+  observability:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.updateStrategy }}
+  updateStrategy:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- with $collector.deploymentUpdateStrategy }}
+  deploymentUpdateStrategy:
+  	{{- toYaml . | nindent 4}}
+  {{- end }}
+  {{- if $collector.volumeMounts }}
+  volumeMounts:
+  {{- with $collector.volumeMounts }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- end }}
+  {{- if $collector.ports }}
+  ports:
+  {{- with $collector.ports }}
+  {{- toYaml . | nindent 4 }}
+  {{- end }}
+  {{- end }}
+  env:
+  {{- with $.Values.extraEnvs }}
+  {{- toYaml . | nindent 2 }}
+  {{- end }}
+  - name: OTEL_K8S_NODE_NAME
+    valueFrom:
+      fieldRef:
+        fieldPath: spec.nodeName
+  - name: OTEL_K8S_NODE_IP
+    valueFrom:
+      fieldRef:
+        fieldPath: status.hostIP
+  - name: OTEL_K8S_NAMESPACE
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: metadata.namespace
+  - name: OTEL_K8S_POD_NAME
+    valueFrom:
+      fieldRef:
+        apiVersion: v1
+        fieldPath: metadata.name
+  - name: OTEL_K8S_POD_UID
```

Comment:
Do we want both the pod uid and name? Thinking it might increase cardinality .
Also, do we really need these env vars as long as we're using the resourcedetection processor and OTEL_RESOURCE_ATTRIBUTES which adds the k8s attributes from the app pod? I'm not sure if that works only for sidecar mode though.

Reply:
we can probs get rid of the UID. I think we def need the name for self telemetry purposes.